app/models/accounting/transaction.rb

```diff
@@ -0,0 +1,37 @@
+# == Schema Information
+#
+# Table name: accounting_transactions
+#
+#  created_at          :datetime         not null
+#  id                  :integer          not null, primary key
+#  qb_transaction_id   :string           not null
+#  qb_transaction_type :string           not null
+#  updated_at          :datetime         not null
+#
+
+class Accounting::Transaction < ActiveRecord::Base
+  attr_writer :qb_object
+
+  delegate :txn_date, :total, :private_note, to: :qb_object
+
+  def qb_object
+    raise ArgumentError, 'qb_object not already set. Use where or all to have it populated for you.' if @qb_object.blank?
+    @qb_object
+  end
+
+  # We need to override where to ensure the qb_object data gets populated when queried.
+  def self.where(**args)
+    transactions = super(args)
+    return transactions if transactions.count < 1
+
+    Accounting::Quickbooks::Fetcher.new(transactions).fetch
+  end
+
+  # We need to override where to ensure the qb_object data gets populated when queried.
+  def self.all
+    transactions = super
+    return transactions if transactions.count < 1
+
+    Accounting::Quickbooks::Fetcher.new(transactions).fetch
+  end
```

Comment:
I don't think overriding these methods is a good idea because it violates the principle of least surprise. They both will now return arrays instead of relations and so they are no longer chainable... Which is fine in itself, I don't see how you could avoid this, but...

Couldn't you add a new class method called `with_qb_objs` that handles the fetch and returns an array, and then do e.g. `Transaction.where(whatever).order(whatever).with_qb_objs`. You would just put a comment on with_qb_objs that says it returns an array and should be at the end of a chain.

Also, please remember to put class methods above instance methods.